src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -74,11 +76,12 @@ public ITaskItem[] FilesWritten
 
         protected override void ExecuteCore()
         {
-            bool writeDevRuntimeConfig = !string.IsNullOrEmpty(RuntimeConfigDevPath);
-
             if (!WriteAdditionalProbingPathsToMainConfig)
             {
-                if (AdditionalProbingPaths?.Any() == true && !writeDevRuntimeConfig)
+                // If we want to generate the runtimeconfig.dev.json file
+                // and we have probing paths to log
+                // BUT the runtimeconfigdevpath is empty, log a warning.
+                if (GenerateRuntimeConfigDevFile && AdditionalProbingPaths?.Any() == true && string.IsNullOrEmpty(RuntimeConfigDevPath))
                 {
                     Log.LogWarning(Strings.SkippingAdditionalProbingPaths);
```

Comment:
If I read it correctly the code will end up throwing in this case. We will log the warning, but then try to write the file - with empty path.